crates/rust-analyzer/src/diagnostics/to_proto.rs

```diff
@@ -87,8 +87,18 @@ fn range(
 ) -> Result<lsp_types::Range> {
     let file_id = from_proto::file_id(snap, &uri)?;
     let line_index = snap.file_line_index(file_id)?;
-    let range =
-        to_proto::range(&line_index, TextRange::new(span.byte_start.into(), span.byte_end.into()));
+    let (start, end) = match line_index.endings {
+        line_index::LineEndings::Unix => (span.byte_start, span.byte_end),
+        line_index::LineEndings::Dos => {
+            let start_crlf_count = (span.line_start as u32).saturating_sub(1);
+            let end_crlf_count = (span.line_end as u32).saturating_sub(1);
```

Comment:
I think this logic is only correct if *every* line uses CRLF, but they might be mixed too

Reply:
i cannot find a good way to get the number of CRLF. maybe we can use the default solution which donot need to calculate the number of CRLF but need more calculation.